### PR TITLE
Add Sticky Header Prop for Table Kit

### PR DESCRIPTION
**Pull request by Ivancandido11**
React Kit Coverage
#### Screens

Screen Shot 2022-05-04 at 2 27 06 PM

https://user-images.githubusercontent.com/84343331/166304855-2b881d65-e8b6-4b8b-a123-89d2cf1936e2.mov

#### Breaking Changes

No, this adds a new prop with a `false` default value so nothing in Nitro will be affected until the prop is passed a `true` value.

#### Runway Ticket URL

[PLAY-67](https://nitro.powerhrg.com/runway/backlog_items/PLAY-67)

#### How to test this

Add the `sticky` prop with a value of `true` and see if the table header sticks when you scroll down.

#### Checklist:

- [ ] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [ ] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [x] **SCREENSHOT** Please add a screen shot or two.
- [x] **SPECS** Please cover your changes with specs.
- [ ] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

### CODE REVIEW

File: playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss [hunk 1]
```diff
@@ -0,0 +1,23 @@
+@import "../../tokens/colors";
+
+[class^="pb_table"] {
+  &.sticky-header {
+    thead {
+      background: $white;
+      position: sticky;
+      top: -40px;
```
Comment (jasoncypret) [inline playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss:8]: I'm not sure about this. Are you setting this because of the nav. We might need a simple test with codesandbox CC @kre8sions

File: playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      z-index: 10;
+    }
+  }
+
+  &.dark {
+    &.sticky-header {
+      thead {
+        background: #0a0527;
```
Comment (jasoncypret) [inline playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss:16]: Lets set this to a playbook token

File: playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss [hunk 3]
```diff
@@ -0,0 +1,23 @@
+@import "../../tokens/colors";
+
+[class^="pb_table"] {
+  &.sticky-header {
+    thead {
+      background: $white;
+      position: sticky;
+      top: -40px;
+      z-index: 10;
```
Comment (jasoncypret) [inline playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss:9]: Why 10?

**Review by jasoncypret — CHANGES_REQUESTED** [review]

File: playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss (hunk 1 above)
Comment (Ivancandido11) [reply playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss:8]: Yea, the padding on the page-content bumps the header down by 40px this makes sure that the header sticks to the right place. And if the padding is different on specific pages I provided examples in the docs to get it to stick to the right place.

File: playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss (hunk 3 above)
Comment (Ivancandido11) [reply playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss:9]: On google searches I saw people using 10 it worked, but Im okay with a different suggestion and learning why I should or should not use 10 here.

**Comment by kre8sions** [comment]
@Ivancandido11 looking for a better way to do this

File: playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss (hunk 3 above)
Comment (jasoncypret) [reply playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss:9]: I'd like to keep it low. z-index is a really tricky thing to deal with. I think setting to match the rich text editor sticky is the way to go.

File: playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss (hunk 1 above)
Comment (jasoncypret) [reply playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss:8]: Ok I'm with you. I don't want to ship a the kit with a fix for our docs. Other peoples sites may not have these issues. Lets fix the problem in playbook and keep top to `0`

[Set the top padding to `0`](https://github.com/powerhome/playbook/blob/master/playbook-website/app/javascript/site_styles/_site-style.scss#L87)
[Add `margin_top: "xl"` here](https://github.com/powerhome/playbook/blob/master/playbook-website/app/views/pages/kit_show.html.erb#L2)

^^^ These updates should allow you to set the top to 0

I'm going to have Jon reach out. We need to add more to the trouble shooting docs. This is a good writeup of dealing with most sticky issues: https://www.designcise.com/web/tutorial/how-to-fix-issues-with-css-position-sticky-not-working

File: playbook/app/pb_kits/playbook/pb_table/docs/_table_sticky.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+React: Use `sticky` on a table to allow the table header to be fixed in place when the user scrolls up and down on the page.
+
+Rails: Pass `sticky: true` to props.
+
+If the table header is not sticking in the right place you will need to pass a inline `top` style to the `thead`.
+React Example: `<thead style={{ top: "-16px" }}>`
+Rails Example: `<thead style="top: -16px">`
+
+Sticky will not work if any parent/ancestor of the sticky element has any of the `overflow` properties set. By specifying a height on the overflowing container, you should be able to make sticky work.
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_table/docs/_table_sticky.md:9]:
```suggestion
## Troubleshooting CSS Problems
Sticky may not work if any parent/ancestor of the sticky element has any of the `overflow` properties set. Additionally, specifying a height on the overflowing container provides measurement for this feature to work properly. In some cases, it may be necessary to set the same parent/ancestor container to `position: static` as well. 
```

File: playbook-website/app/views/pages/kit_show.html.erb [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <div class="pb--kit-show <%=@kit%>-kit">
-  <%= pb_rails("title", props: { text: pb_kit_title(@kit), tag: "h1", size: 1 }) %>
+  <%= pb_rails("title", props: { text: pb_kit_title(@kit), tag: "h1", size: 1, margin: "xl" }) %>
```
Comment (thestephenmarshall) [inline playbook-website/app/views/pages/kit_show.html.erb:2]: Why was this added?

**Review by thestephenmarshall — COMMENTED** [review]
I dropped in the suggestion and have a question.

File: playbook-website/app/views/pages/kit_show.html.erb (hunk 5 above)
Comment (Ivancandido11) [reply playbook-website/app/views/pages/kit_show.html.erb:2]: Thats was @jasoncypret suggestion so we could use `top: 0`.

File: playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss (hunk 3 above)
Comment (thestephenmarshall) [reply playbook/app/pb_kits/playbook/pb_table/styles/_sticky_header.scss]: [Use this as an example](https://github.com/powerhome/playbook/blob/PLAY-67-add-sticky-headers-for-table-kit/playbook/app/pb_kits/playbook/pb_rich_text_editor/_rich_text_editor.scss#L84)

**Review by jasoncypret — APPROVED** [review]
LGTM 🚀 Great Job @Ivancandido11

**Review by thestephenmarshall — APPROVED** [review]
